tests/EStimLibrary.UnitTests/Extensions/SpatialModel/StringHierarchy/StringHierarchySpecTests.cs

```diff
@@ -0,0 +1,552 @@
+    /// StringHierarchySpec((string[] RegionSet, string[] ModifierSet) tuple) --- DONE
+    /// ParseFullSpec(string fullSpec) --- DONE
+    /// ParseRegionSpec(string regionSpec) --- DONE
+    /// ParseModifierSpec(string modifierSpec) --- DONE
+    /// JoinFullSpec(string[] regionSet, string[] modifierSet) --- DONE
+    /// JoinRegionSet(string[] regionSet) --- DONE
+    /// JoinModifierSet(string[] modifierSet) --- DONE
+    /// TryParseOptionedRegionName(string optionedRegionName, out string baseName, out string options) --- DONE
+    /// RegionSetOverlaps(StringHierarchySpec other, out string[] sharedRegionSet) --- DEFUNCT
+    /// ModifiersAllowOverlap(StringHierarchySpec other, out string[] commonModifiers) --- DEFUNCT
+    /// TryGetOverlap(StringHierarchySpec other, out string overlappingRegion, out bool contains) --- MOVE TO BodyModel TESTING
+    /// Equals(StringHierarchySpec? other) --- DONE
+    /// GetHashCode() --- DONE
+    /// ToString() --- DONE
+
+    /// <summary>
+    /// Test the static example spec is correct.
+    /// </summary>
+    [Fact]
+    public void ExamplePath_ShouldBeCorrect()
+    {
+        Assert.Equal("option1 region1, no-option-region2, ... | modifier, ...",
+            StringHierarchySpec.ExamplePath);
+    }
+
+    /// <summary>
+    /// Test the constructor StringHierarchySpec that takes the full string 
+    /// spec, where the input field is not empty.
+    /// </summary>
+    [Theory]
+    // Case with regions and modifiers
+    [InlineData("region1, region2 | modifier1, modifier2",
+                new[] { "region1", "region2" },
+                new[] { "modifier1", "modifier2" })]
+    // Case with only one region
+    [InlineData("region1",
+                new[] { "region1" },
+                new string[0])]
+    // Case sensitivity
+    [InlineData("Region1, REGION2 | Modifier1, MODIFIER2",
+                new[] { "region1", "region2" },
+                new[] { "modifier1", "modifier2" })]
+    // Single region and modifier
+    [InlineData("region1 | modifier1",
+                new[] { "region1" },
+                new[] { "modifier1" })]
+    public void ConstructorFullSpec_ValidInputs_ShouldInit(string fullSpec, 
+        string[] expectedRegions, string[] expectedModifiers)
+    {
+        // Build the spec and make sure sets parsed correctly
+        var stringHierarchySpec = new StringHierarchySpec(fullSpec);
+        Assert.Equal(expectedRegions, stringHierarchySpec.RegionSet);
+        Assert.Equal(expectedModifiers, stringHierarchySpec.ModifierSet);
+
+        // Make sure partial string specs are correct
+        var specHalves = fullSpec.Split(
+            StringHierarchySpec.REGIONS_MODIFIERS_DELIMITER);
+        Assert.Equal(specHalves[0].Trim(), stringHierarchySpec.RegionSpec);
+        Assert.Equal(specHalves[1].Trim(), stringHierarchySpec.ModifierSpec);
+    }
+
+    /// <summary>
+    /// Test the constructor StringHierarchySpec(string fullSpec), where the
```

Comment:
Other style things I started fixing but will let you finish (Ik nit-picky and maybe tedious, but I'm being annoying about it now so we get in the habit of doing it in the first place): try to have lines be <= 80 characters and enter/indent after that. This is to help readability.

If [InlineData()] cases are multi-line too, just put the comments in a line on top of them rather than at the end of the last line. That way can read top-down and know what's coming.